### PR TITLE
Fixes issue #13227 

### PR DESCRIPTION
**Pull request by atishgoswami**
## Description
Recently Viewed contains wrong product url with category path

This PR will:

- Generate the proper product url

## Fixed Issues (if relevant)

#13227 

## Contribution checklist

- [x] Pull request has a meaningful description of its purpose
- [x] All commits are accompanied by meaningful commit messages
- [ ] All new or changed code is covered with unit/integration tests (if applicable)
- [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22650) <br/>All committers have signed the CLA.

**Comment by atishgoswami** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @atishgoswami. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @atishgoswami, here is your [new Magento instance](https://pr-22650.instances.magento-community.engineering).
Admin access: [https://pr-22650.instances.magento-community.engineering/admin](https://pr-22650.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

File: app/code/Magento/Catalog/Model/Product/Url.php [hunk 1]
```diff
@@ -162,11 +162,8 @@ public function getUrl(\Magento\Catalog\Model\Product $product, $params = [])
                     \Magento\Store\Model\ScopeInterface::SCOPE_STORE
                 );
 
-                if ($categoryId) {
-                    $filterData[UrlRewrite::METADATA]['category_id'] = $categoryId;
-                } elseif (!$useCategories) {
-                    $filterData[UrlRewrite::METADATA]['category_id'] = '';
-                }
+                $filterData[UrlRewrite::METADATA]['category_id']
```
Comment (rogyar) [inline app/code/Magento/Catalog/Model/Product/Url.php]: Hi @atishgoswami. Thank you for your contribution.
That's a very good point. If the corresponding configuration is not enabled, the system should not take into account a category ID

**Comment by rogyar** [comment]
That's a very good point. If the corresponding configuration is not enabled, the system should not take into account a category ID

**Comment by atishgoswami** [comment]
Seems like were couple unrelated test that seems to be broken. 
Is there a way to re-trigger the travis build?

**Comment by rogyar** [comment]
Hi @atishgoswami. Sorry for the delayed answer. I believe the failing integration tests are related to the current PR. The system expects URL to have a category in the path but gets direct URL to the project. We need to adjust the tests correspondingly.

**Review by orlangur — CHANGES_REQUESTED** [review]
Nice! Your PR already includes my notes from https://github.com/magento/magento2/pull/22322/files#r275178147 :)

Please remove redundant parentheses, adopt existing integration tests and add a new one corresponding to described case.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

**Comment by atishgoswami** [comment]
Sure will do @rogyar and @orlangur

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-5300](https://jira.corp.magento.com/browse/ENGCOM-5300) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**